view/base/web/js/model/address_validation_core.js

```diff
@@ -93,8 +89,46 @@ function (ko, $) {
             }
         },
 
+        isSuggestedAddress: function (address) {
+            var self = this;
+            var isSuggested = false;
+            var suggestedAddresses = this.suggestedAddresses();
+            suggestedAddresses.forEach(function(suggestedAddress, index) {
+                if (self.addressMatches(self.formatAddress(address), self.formatSuggestedAddress(suggestedAddress.address))) {
+                    isSuggested = true;
+                }
+            });
```

Comment:
I'm not sure how large `suggestedAddresses` could be, so this may not matter, but what about returning early if a match is found? I think you'd have to use a `for` loop instead of `forEach` though.

Reply:
I like the suggestion, and you are right about having to use a for loop.  However this array should only have have 2 or 3 entries. This whole feature is going to be refactored in the future, so I think for now the current approach is satisfactory.